include/envoy/udp/hash_policy.h

```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+#include "envoy/network/address.h"
+
+#include "absl/types/optional.h"
+
+namespace Envoy {
+namespace Udp {
+/**
+ * Hash policy for UDP transport layer protocol.
+ */
+class HashPolicy {
+public:
+  virtual ~HashPolicy() = default;
+
+  /**
+   * @param downstream_address is the address of the peer client.
+   * @return absl::optional<uint64_t> an optional hash value to route on. A hash value might not be
+   * returned if for example the downstream address has a nullptr.
+   */
+  virtual absl::optional<uint64_t>
+  generateHash(const Network::Address::InstanceConstSharedPtr& downstream_address) const PURE;
```

Comment:
Sorry I mean `const Network::Address::Instance&`. I don't think it's possible for the address to be null.

Reply:
Ok. it has been changed.